### PR TITLE
add move_cursor_{left,right} to TermWrite

### PR DESCRIPTION
**Pull request by MovingtoMars**

### CODE REVIEW

File: src/control.rs [hunk 1]
```diff
@@ -43,6 +43,32 @@ pub trait TermWrite {
...
+    /// Move the cursor `num` spaces to the left.
+    fn move_cursor_left(&mut self, num: u32) -> io::Result<usize> {
+        if num > 0 {
+            self.csi(&[b'0' + (num / 10000) as u8,
+                       b'0' + (num / 1000) as u8 % 10,
+                       b'0' + (num / 100) as u8 % 10,
+                       b'0' + (num / 10) as u8 % 10,
+                       b'0' + num as u8 % 10,
+                       b'C'])
+        } else {
```
Comment (ticki) [inline src/control.rs:55]: No reason for this special case.

File: src/control.rs [hunk 2]
```diff
@@ -43,6 +43,32 @@ pub trait TermWrite {
...
+    /// Move the cursor `num` spaces to the right.
+    fn move_cursor_right(&mut self, num: u32) -> io::Result<usize> {
+        if num > 0 {
+            self.csi(&[b'0' + (num / 10000) as u8,
+                       b'0' + (num / 1000) as u8 % 10,
+                       b'0' + (num / 100) as u8 % 10,
+                       b'0' + (num / 10) as u8 % 10,
+                       b'0' + num as u8 % 10,
+                       b'D'])
+        } else {
```
Comment (ticki) [inline src/control.rs:68]: Here too.

File: src/control.rs (hunk 1 above)
Comment (MovingtoMars) [inline src/control.rs:55]: On my terminal, going `csi("0C")` still moves the cursor right by one, so I have to special case at call site.

Comment (ticki) [inline src/control.rs:55]: Oh, well. Then.

**Comment by ticki** [comment]
:+1:

**Pull request merged**